src/components/Cluster/ClusterDetail/AddNodePool/AddNodePoolAZSelection.tsx

```diff
@@ -0,0 +1,255 @@
+  }
+
+  return (
+    <div {...rest}>
+      <AZSelectionLabel>Availability Zones selection</AZSelectionLabel>
+      <StyledPanel
+        expanded={value === AvailabilityZoneSelection.Automatic}
+        onToggle={onToggleFakeCallback}
+      >
+        <AddNodePoolAZSelectionCheckbox
+          onChange={onChange}
+          value={value}
+          npID={npID}
+          label='Automatic'
+          type={AvailabilityZoneSelection.Automatic}
+        />
+        <StyledPanelCollapse>
+          <AZSelectorWrapper>
+            <p>Number of availability zones to use:</p>
+            <AvailabilityZonesParser
+              min={minNumOfZones}
+              max={maxNumOfZones}
+              defaultValue={defaultNumOfZones}
+              zones={allZones}
+              updateAZValuesInParent={onUpdateZones(
+                AvailabilityZoneSelection.Automatic
+              )}
+              isLabels={false}
+            />
+          </AZSelectorWrapper>
+          <p>{automaticAZSelectionMessage}</p>
+        </StyledPanelCollapse>
+      </StyledPanel>
+      <StyledPanel
+        expanded={value === AvailabilityZoneSelection.Manual}
+        onToggle={onToggleFakeCallback}
+      >
+        <AddNodePoolAZSelectionCheckbox
+          onChange={onChange}
+          value={value}
+          npID={npID}
+          label='Manual'
+          type={AvailabilityZoneSelection.Manual}
+        />
+        <StyledPanelCollapse>
+          <p>
+            You can select up to {maxNumOfZones} availability zones to make use
+            of.
+          </p>
+          <AZSelectorWrapper>
+            <ManualAZSelector>
+              <AvailabilityZonesParser
+                min={minNumOfZones}
+                max={maxNumOfZones}
+                defaultValue={2}
+                zones={allZones}
+                updateAZValuesInParent={onUpdateZones(
+                  AvailabilityZoneSelection.Manual
+                )}
+                isLabels={true}
+              />
+            </ManualAZSelector>
+
+            {manualAZSelectionErrorMessage && (
+              <ErrorMessage>{manualAZSelectionErrorMessage}</ErrorMessage>
+            )}
+          </AZSelectorWrapper>
+        </StyledPanelCollapse>
+      </StyledPanel>
+
+      {provider === Providers.AZURE && (
+        <StyledPanel
+          expanded={value === AvailabilityZoneSelection.NotSpecified}
+          onToggle={onToggleFakeCallback}
+        >
+          <AddNodePoolAZSelectionCheckbox
+            onChange={onChange}
+            value={value}
+            npID={npID}
+            label='Not specified'
+            type={AvailabilityZoneSelection.NotSpecified}
+          />
+          <StyledPanelCollapse>
+            <p>
+              By not specifying an availability zone, Azure will select a zone
+              by itself, where the requested VM size has the best availability.
```

Comment:
Avoiding abbreviations where ever we can

```suggestion
              by itself, where the requested virtual machine size has the best availability.
```